backend/services/deployments/store/mongo/datastore_mongo.go

```diff
@@ -1284,6 +1284,137 @@ func (db *DataStoreMongo) ListImages(
 	return images, int(count), nil
 }
 
+/*
+// This could be used in the future to support sorting
+
+	func getImagesSortFieldAndOrder(filt *model.ImageFilter) (string, int) {
+		if filt != nil && filt.Sort != "" {
+			sortParts := strings.SplitN(filt.Sort, ":", 2)
+			if len(sortParts) == 2 &&
+				(sortParts[0] == "name" ||
+					sortParts[0] == "modified" ||
+					sortParts[0] == "artifacts_count" ||
+					sortParts[0] == "tags") {
+				sortField := sortParts[0]
+				sortOrder := 1
+				if sortParts[1] == model.SortDirectionDescending {
+					sortOrder = -1
+				}
+				return sortField, sortOrder
+			}
+		}
+		return "", 0
+	}
+*/
+func (db *DataStoreMongo) ListImagesV2(
+	ctx context.Context,
+	filt *model.ImageFilter,
+) ([]*model.Image, error) {
+	database := db.client.Database(mstore.DbFromContext(ctx, DatabaseName))
+	collImg := database.Collection(CollectionImages)
+
+	filters := bson.M{}
+
+	if filt != nil {
+
+		var nameFilter []interface{}
+		// Check for exact names
+		for _, name := range filt.ExactNames {
+			nameFilter = append(nameFilter, bson.M{
+				StorageKeyImageName: name,
+			})
+		}
+		// Check for prefixes
+		for _, name := range filt.NamePrefixes {
+			nameFilter = append(nameFilter, bson.M{
+				StorageKeyImageName: bson.M{
+					"$regex": primitive.Regex{
+						Pattern: "^" + regexp.QuoteMeta(name) + ".*",
+						Options: "i",
+					},
+				},
+			})
+		}
+		if len(nameFilter) > 0 {
+			filters["$or"] = nameFilter
+		}
```

Comment:
(suggestion) Since the options are mutually exclusive this could be simplified.
```suggestion
	if len(filt.ExactNames) > 0 {
		filters[StorageKeyName] = bson.M{"$in": filt.ExactNames}
	} else if filt.NamePrefix != "" {
		filters[StorageKeyName] = bson.M{"$regex": "^" + regexp.QuoteMeta(filt.NamePrefix)}	
	}
```